### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by sdjustin**
Updating node version in CONTRIBUTING.md doc. Requires at a node v14 to run `yarn start:docs`

### CODE REVIEW

**Review by ktabors — APPROVED** [review]
We know there are some versions of v12 that work, like v12.22.x, but telling everyone to use v14 will improve the dev experience.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ If you are looking for place to start, consider the following options:
 
 ## Developing
 When you are ready to start developing you can clone the repo and start storybook.
-Make sure you have the following requirements installed: [node](https://nodejs.org/) (v11.0.0+) and [yarn](https://yarnpkg.com/en/) (v1.22.0+)
+Make sure you have the following requirements installed: [node](https://nodejs.org/) (v14.0.0+) and [yarn](https://yarnpkg.com/en/) (v1.22.0+)
```
Comment (snowystinger) [inline CONTRIBUTING.md:61]:
```suggestion
Make sure you have the following requirements installed: [node](https://nodejs.org/) (v14.15.0+) and [yarn](https://yarnpkg.com/en/) (v1.22.0+)
```
the node engine won't work with any 14's below 14.15
I just checked and 14.15 does work

**Review by snowystinger — APPROVED** [review]
thanks for the PR!

**Review by reidbarber — APPROVED** [review]
Thanks!

**Pull request merged**